gems/sorbet-runtime/lib/types/enum.rb

```diff
@@ -298,6 +299,13 @@ def self.fully_initialized?
     @fully_initialized ||= false
   end
 
+  # Maintains the order in which values are defined
+  sig {params(instance: T.untyped).void}
+  def self._register_instance(instance)
+    @values ||= []
+    @values << T.cast(instance, T.attached_class)
```

Comment:
What are you doing with `T.cast` here? `instance` is untyped so you don't need this cast statically.
Also `T.attached_class` isn't checked at runtime, so you don't need this cast at runtime either.

Reply:
I intended this to,
1. Compensate for the `T.untyped` in the sig, which is present because of `T.attached_class` not working with `params`.
1. Make it clear to a _reader_ that it's not really untyped.

Should I remove?